classes/Menu.php

```diff
@@ -151,7 +152,12 @@ public function generateReferences($page)
         $iterator = function($items) use ($currentUrl, &$iterator, $activeMenuItem) {
             $result = [];
 
+            $locale = App::getLocale();
             foreach ($items as $item) {
+                if ($title = array_get($item->viewBag, "localeTitle.$locale")) {
+                    $item->title = $title;
+                }
+
```

Comment:
@mjauvin This should probably be done through the `pages.menu.referencesGenerated` event in the Translate plugin itself.

Reply:
Same thing as above.